### PR TITLE
Docs postgresql 9.4 merge ga

### PR DESCRIPTION
**Pull request by litzell**
Updates GPDB 6 DITA docs with changes to sgml files in Postgres 9.4 merge. This is part 2. 

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml [hunk 1]
```diff
@@ -1,54 +1,109 @@
...
+      <p>Greenplum Database creates a temporary variable of data type <varname>stype</varname> to
+        hold the current internal state of the aggregate function. At each input row, the aggregate
+        argument values are calculated and the state transition function is invoked with the current
+        state value and the new argument values to calculate a new internal state value. After all
+        the rows have been processed, the final function is invoked once to calculate the aggregate
+        return value. If there is no final function then the ending state value is returned
+        as-is.</p><p>You can specify <codeph>COMBINEFUNC</codeph> as a method for optimizing aggregate execution. By
+        specifying <codeph>COMBINEFUNC</codeph>, the aggregate can be executed in parallel on
+        segments first and then on the master. When a two-level execution is performed,
           <codeph>SFUNC</codeph> is executed on the segments to generate partial aggregate results,
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: wondering if we should be using the function name (i.e. <varname>dfunc</varname>) vs the all caps keyword name?  in any case, we should be consistent throughout, and use varname/codeph consistently as well.  see below.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml [hunk 2]
```diff
@@ -62,24 +117,56 @@ ffunc( internal-state ) ---&gt; aggregate-value</codeblock><p>You
...
         example, the final function for <codeph>avg</codeph> returns <codeph>NULL</codeph> when it
-        sees there were zero input rows.</p><p>Single argument aggregate functions, such as min or
+        sees there were zero input rows.</p>
+    <p>
+      Sometimes it is useful to declare the final function as taking not just
+      the state value, but extra parameters corresponding to the aggregate's
+      input values.  The main reason for doing this is if the final function
+      is polymorphic and the state value's data type would be inadequate to
+      pin down the result type.  These extra parameters are always passed as
+      NULL (and so the final function must not be strict when
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: codeph NULL?

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml [hunk 3]
```diff
@@ -90,39 +177,175 @@ ffunc( internal-state ) ---&gt; aggregate-value</codeblock><p>You
...
+      <parml>
+        <plentry>
+          <pt><varname>name</varname></pt>
+          <pd>The name (optionally schema-qualified) of the aggregate function to create.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>argmode</varname></pt>
+          <pd>The mode of an argument: <codeph>IN</codeph> or <codeph>VARIADIC</codeph>. (Aggregate
+            functions do not support <codeph>OUT</codeph> arguments.) If omitted, the default is
+              <codeph>IN</codeph>. Only the last argument ca be marked
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: ca -> **can**

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml [hunk 4]
```diff
@@ -90,39 +177,175 @@ ffunc( internal-state ) ---&gt; aggregate-value</codeblock><p>You
...
+              <varname>state_data_type</varname>. The planner uses this value to estimate the memory
+            required for a grouped aggregate query. Large values of this parameter discourage use of
+            hash aggregation.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>ffunc</varname></pt>
+          <pd>The name of the final function called to compute the aggregate result after all input
+            rows have been traversed. The function must take a single argument of type
               <codeph>state_data_type</codeph>. The return data type of the aggregate is defined as
             the return type of this function. If <codeph>ffunc</codeph> is not specified, then the
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: varname ffunc?

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 5]
```diff
@@ -308,7 +309,11 @@ SELECT foo();</codeblock><p>In
                         dynamically loadable object, and <varname>link_symbol</varname> is the name
                         of the function in the C language source code. If the link symbol is
                         omitted, it is assumed to be the same as the name of the SQL function being
-                        defined. It is recommended to locate shared libraries either relative to
+                        defined. The C names
+                        of all functions must be different, so you must give overloaded C
```
[the remark this answers is not in the document]
Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: I think overloaded is the right word, but I think it's the SQL functions that are overloaded, not the C functions, so I changed 'C' to 'SQL'

File: gpdb-doc/dita/ref_guide/sql_commands/DROP_USER.xml [hunk 6]
```diff
@@ -11,9 +11,10 @@
     </section>
     <section id="section3">
       <title>Description</title>
-      <p><codeph>DROP USER</codeph> is an alias for <codeph><xref href="DROP_ROLE.xml#topic1"
-            type="topic" format="dita"/></codeph>. See <codeph><xref href="DROP_ROLE.xml#topic1"
-            type="topic" format="dita"/></codeph> for more information.</p>
+      <p><codeph>DROP USER</codeph> is simply an alternate spelling of <codeph><xref
```
[the remark this answers is not in the document]
Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/DROP_USER.xml]: Agree. And 'simply' is simply unnecessary.

File: gpdb-doc/dita/ref_guide/sql_commands/REASSIGN_OWNED.xml [hunk 7]
```diff
@@ -12,7 +12,11 @@
     <section id="section3">
       <title>Description</title>
       <p><codeph>REASSIGN OWNED</codeph> changes the ownership of database objects owned by any of
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/REASSIGN_OWNED.xml]: zombie

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml (hunk 1 above)
Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: Went with the lowercase, italic function params.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml [hunk 8]
```diff
@@ -1,54 +1,109 @@
...
+  or the old syntax
+  
+  CREATE AGGREGATE <varname>name</varname> (
+    BASETYPE = <varname>base_type</varname>,
+    SFUNC = <varname>sfunc</varname>,
+    STYPE = <varname>state_data_type</varname>
+    [ , SSPACE = <varname>state_data_size</varname> ]
+    [ , FINALFUNC = <varname>ffunc</varname> ]
+    [ , FINALFUNC_EXTRA ]
+    [ , COMBINEFUNC = <varname>combinefunc</varname> ]
```
[the remark this answers is not in the document]
Comment (dyozie) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: Looks like these weren't added until 9.6.  @litzell - wondering if maybe you took the latest source from the master branch when updating this page?

[the remark this answers is not in the document]
Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: It matches the online help in 6.0.0-alpha.0+dev.16123.gd0839b2 as well as the 9.4 PR I worked from. Perhaps it got pulled in early?

**Comment by dyozie** [comment]
True - it wouldn't be the first time we pulled something from an even-later
release.  If it's in the Beta I'm comfortable leaving it in the docs as
well.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml [hunk 9]
```diff
@@ -187,8 +416,12 @@ Aggregate (cost=1.10..1.11 rows=1 width=32)  
       width=32)
      -&gt; Aggregate (cost=1.04..1.05 rows=1 width=32)
        -&gt; Seq Scan on t1 (cost=0.00..1.03 rows=2 width=8)
- (4 rows)</codeblock></section><section id="section8"><title>Compatibility</title><p><codeph>CREATE AGGREGATE</codeph> is a Greenplum Database language extension.
-The SQL standard does not provide for user-defined aggregate functions.</p></section><section id="section9"><title>See Also</title><p><codeph><xref href="ALTER_AGGREGATE.xml#topic1" type="topic" format="dita"/></codeph>,
+ (4 rows)</codeblock></section><section id="section8"><title>Compatibility</title><p>The SQL standard does not provide for user-defined aggregate functions.</p>
+      <p>The Greenplum Database <codeph>CREATE AGGREGATE</codeph> command is based on the PostgreSQL
+        command. Greenplum Database adds the <codeph><varname>combinefunc</varname></codeph>,
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_AGGREGATE.xml]: I'm wondering if this statement is misleading.  Did Greenplum actually create and contribute combinefunc, serialfunc, and deserialfunc upstream to postgres, or did we just pull them from a later version of postgres.  They do seem customized for MPP use, so maybe this is correct as is.

**Pull request merged**